### DISCUSSION TITLE
Does declaring an object property as unknown make it optional?

### DISCUSSION THREAD

**Discussion by aryzing** [open-circle/valibot#677] [Q&A]
Moved to https://discord.com/channels/1252985447273992222/1255508856608522341/1255508856608522341

**Comment by harshagarwalnyu** [comment]
No — `unknown()` and `optional()` are independent concepts in Valibot:

- **`v.unknown()`** — the property is **required** but accepts **any value** (no type constraint). It must be present in the input.
- **`v.optional(schema)`** — the property may be **absent** from the input (or explicitly `undefined`). The wrapped schema only runs when the value is present.

So `{ key: v.unknown() }` requires `key` to exist but accepts any value for it. If `key` is missing, parsing fails.

**To make a property accept any value AND be optional:**
```ts
import * as v from 'valibot';

const Schema = v.object({
  required_any: v.unknown(),           // required, any value
  optional_typed: v.optional(v.string()), // optional, must be string if present
  optional_any: v.optional(v.unknown()),  // optional, any value if present
});

// All three parse successfully:
v.parse(Schema, { required_any: 42 });
v.parse(Schema, { required_any: 42, optional_typed: 'hello' });
v.parse(Schema, { required_any: 42, optional_any: { nested: true } });

// This fails — required_any is missing:
v.parse(Schema, {});
```

**Summary:**
| Schema | Required? | Type constraint |
|--------|-----------|-----------------|
| `v.unknown()` | Yes | None (any value) |
| `v.optional(v.string())` | No | `string` if present |
| `v.optional(v.unknown())` | No | None (any value if present) |
| `v.nullable(v.string())` | Yes | `string` or `null` |
